src/value.c

```diff
@@ -539,6 +539,10 @@ int SIValue_Compare(const SIValue a, const SIValue b, int *disjointOrNull) {
 		case T_BOOL:
 			return SAFE_COMPARISON_RESULT(a.longval - b.longval);
 		case T_DOUBLE:
+			if(isnan(a.doubleval) || isnan(b.doubleval)) {
+				if(disjointOrNull) *disjointOrNull = COMPARED_NAN;
+			}
+
 			return SAFE_COMPARISON_RESULT(a.doubleval - b.doubleval);
```

Comment:
Do we want to perform `SAFE_COMPARISON_RESULT` if either `a` or `b` are `NaN` ?

Reply:
it will be calculated to nan with is the expected result